tests/rest/client/test_transactions.py

```diff
@@ -34,11 +38,13 @@ def setUp(self) -> None:
         self.hs.get_auth = Mock()
         self.cache = HttpTransactionCache(self.hs)
 
-        self.mock_http_response = (HTTPStatus.OK, "GOOD JOB!")
+        self.mock_http_response = (HTTPStatus.OK, {"result": "GOOD JOB!"})
         self.mock_key = "foo"
 
     @defer.inlineCallbacks
-    def test_executes_given_function(self):
+    def test_executes_given_function(
+        self,
+    ) -> Generator["defer.Deferred[Any]", object, None]:
```

Comment:
I think this signature means that

- `res: Any`?
- no constraints on the type of the thing being yielded

That's ever so slightly sad, but I don't think there's a good  way to get mypy to check this short of using proper `async defs`. (And I think that is a little awkward to do in tests?)

Reply:
It unfortunately needs to match what `deferred.inlineCallbacks` wants, I think. And the type on that isn't super accurate IIRC?